src/algebra/big_operators/basic.lean

```diff
@@ -395,6 +377,35 @@ begin
     simp only [prod_insert H, prod_mul_distrib, ih] }
 end
 
+@[to_additive]
+lemma prod_product_left {s : finset γ} {t : finset α} {f : γ×α → β} :
```

Comment:
I'm inclined to keep calling this `prod_product`. I agree with Eric that this is the "natural" statement.